python/setup.py

```diff
@@ -480,47 +455,14 @@ def has_ext_modules(foo):
 
 
 setup(
-    name='pyarrow',
     packages=packages,
-    zip_safe=False,
-    package_data={'pyarrow': ['*.pxd', '*.pyx', 'includes/*.pxd']},
-    include_package_data=True,
     exclude_package_data=exclude_package_data,
     distclass=BinaryDistribution,
     # Dummy extension to trigger build_ext
     ext_modules=[Extension('__dummy__', sources=[])],
     cmdclass={
         'build_ext': build_ext
     },
-    use_scm_version={
-        'root': os.path.dirname(setup_dir),
-        'parse': parse_git,
-        'write_to': os.path.join(scm_version_write_to_prefix,
-                                 'pyarrow/_generated_version.py'),
-        'version_scheme': guess_next_dev_version
-    },
-    setup_requires=['setuptools_scm', 'cython >= 0.29.31'] + setup_requires,
-    install_requires=install_requires,
     tests_require=['pytest', 'pandas', 'hypothesis'],
```

Comment:
I don't know if anyone is relying on this? 
Given one shouldn't run the tests through setuptools (which is also deprecated, I think), I think we can remove this as well (same for `test_suite` below)

We could add a tests "extra" in pyproject.toml with those dependencies if we want (although strictly speaking pandas is not actually a required test dependency, we will skip tests if pandas is not installed. So if we do add this extra, we should use an up to date list of actual test dependencies)

Reply:
> So if we do add this extra, we should use an up to date list of actual test dependencies)

So rather than it comprehensively installing all of the dependencies that any test might rely on, you only want the non-optional ones?

Reply:
I've pushed a change removing those. Will do a final CI run to validate all is good!